### DISCUSSION TITLE
Reproducing a command using cloning and composition

### DISCUSSION THREAD

**Discussion by fuglede** [dlemstra/magick-wasm#66] [Q&A] [answered]
I'm trying to reproduce the following command:

```
convert input.png -colorspace gray \( +clone -blur 0x1 \) +swap -compose divide -composite -linear-stretch 5%x0% output.png
```

As far as I can tell, `linear-stretch` isn't available, but `contrast-stretch` is. What I've tried is the below (and I've tried with `DiviceDst` instead of `DivideSrc` as well), but the result doesn't match what I get on the command line:

```
        ImageMagick.read(array, image => {
          image.colorSpace = ColorSpace.Gray
          image.clone(clone => {
            clone.blur(0, 1)
            image.composite(clone, CompositeOperator.DivideSrc)
            image.contrastStretch(new Percentage(5), new Percentage(0))
            image.rotate(1.5)
            const canvas = document.getElementById('canvas')
            image.writeToCanvas(canvas)
          })
```

What am I doing wrong?

Edit: Actually, it's close enough that it's probably just the difference between the stretching methods, so let me not worry more about that. I'd delete the thread entirely if that was possible. :)

**Comment by dlemstra** [comment, marked as answer]
Support for linear-stretch will become available in the next release.

**Reply by fuglede** [reply]
Thanks for the heads-up, that's great!
